opensrp-app/src/test/java/org/smartregister/view/fragment/BaseRegisterFragmentTest.java

```diff
@@ -0,0 +1,208 @@
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.Robolectric;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.util.ReflectionHelpers;
+import org.smartregister.BaseUnitTest;
+import org.smartregister.Context;
+import org.smartregister.R;
+import org.smartregister.cursoradapter.RecyclerViewPaginatedAdapter;
+import org.smartregister.view.activity.SecuredNativeSmartRegisterActivity;
+import org.smartregister.view.contract.BaseRegisterFragmentContract;
+
+/**
+ * Created by ndegwamartin on 2020-04-28.
+ */
+public class BaseRegisterFragmentTest extends BaseUnitTest {
+
+    private BaseRegisterFragment baseRegisterFragment;
+
+    @Mock
+    private LayoutInflater layoutInflater;
+
+    @Mock
+    private ViewGroup container;
+
+    @Mock
+    private Bundle bundle;
+
+    private Activity activity;
+
+    @Mock
+    private EditText searchView;
+
+    @Mock
+    private Context opensrpContext;
+
+    @Mock
+    private TextWatcher textWatcher;
+
+    @Mock
+    private View.OnKeyListener hideKeyboard;
+
+    @Mock
+    private View searchCancelView;
+
+    @Mock
+    protected BaseRegisterFragmentContract.Presenter presenter;
+
+    @Mock
+    public RecyclerViewPaginatedAdapter clientAdapter;
+
+    @Captor
+    private ArgumentCaptor<Boolean> qrCodeArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> openSRPIdArgumentCaptor;
+
+    @Captor
+    private ArgumentCaptor<String> stringArgumentCaptor;
+
+    @Before
+    public void setUp() {
+
+        MockitoAnnotations.initMocks(this);
+        baseRegisterFragment = Mockito.mock(BaseRegisterFragment.class, Mockito.CALLS_REAL_METHODS);
+        ReflectionHelpers.setField(baseRegisterFragment, "presenter", presenter);
+        ReflectionHelpers.setField(baseRegisterFragment, "clientAdapter", clientAdapter);
+
+        activity = Robolectric.buildActivity(AppCompatActivity.class).get();
+    }
+
+    @Test
+    public void assertFragmentInstantiatesCorrectly() {
+
+        Assert.assertNotNull(baseRegisterFragment);
+    }
+
+    @Test
+    public void assertGetNavBarOptionsProviderNotNull() {
+        SecuredNativeSmartRegisterActivity.NavBarOptionsProvider provider = baseRegisterFragment.getNavBarOptionsProvider();
+        Assert.assertNotNull(provider);
+    }
+
+    @Test
+    public void assertGetNavBarOptionsProviderReturnsCorrectValueFormSearchHint() {
+        SecuredNativeSmartRegisterActivity.NavBarOptionsProvider provider = baseRegisterFragment.getNavBarOptionsProvider();
+
+        Mockito.doReturn(opensrpContext).when(baseRegisterFragment).context();
+        Mockito.doReturn(RuntimeEnvironment.application.getResources().getString(R.string.search_hint)).when(opensrpContext).getStringResource(R.string.search_hint);
+
+        String hint = RuntimeEnvironment.application.getResources().getString(R.string.search_hint);
+        Assert.assertEquals(hint, provider.searchHint());
+    }
+
+    @Test
+    @Ignore
```

Comment:
Create an issue to fix this test & track it

Reply:
Done, [Issue 509](https://github.com/OpenSRP/opensrp-client-core/issues/509)